frontend/packages/dsco/CHANGELOG-pre1.md

```diff
@@ -0,0 +1,167 @@
+
+- created `hooks` folder
+- added `useBodyScrollLock`, `useDocumentKeydown`, `useEscapeKeyHandler`, `useFocusTrap` hooks
+- updated Readme.md
+
+## [0.4.0](https://github.com/code-dot-org/code-dot-org/pull/62023)
+
+- created `common/styles/variables.scss` file
+- added `DropdownFormFieldRelatedProps` type constant
+- minor `Changelog` refactor by prettier
+
+## [0.3.4](https://github.com/code-dot-org/code-dot-org/pull/60916)
+
+- updated colors.css values
+- removed utility colors
+
+## [0.3.2](https://github.com/code-dot-org/code-dot-org/pull/59328)
+
+- added `updatePositionedElementStyles` and `calculatePositionedElementStyles` helper functions
+- minor types update
+
+## [0.3.1](https://github.com/code-dot-org/code-dot-org/pull/59609)
+
+- added colors.css which implements all the Semantic and Utility colors
+
+## [0.3.0](https://github.com/code-dot-org/code-dot-org/pull/59500)
+
+- added primitiveColors.css
+
+## [0.2.1](https://github.com/code-dot-org/code-dot-org/pull/54064)
+
+- minor README.md updates
+
+## [0.2.0](https://github.com/code-dot-org/code-dot-org/pull/53657)
+
+- added `styles/mixins.scss`
+- added README.md
+- added CHANGELOG.md
+
+## [0.1.0](https://github.com/code-dot-org/code-dot-org/pull/52753)
+
+- added `constants.ts`
+- added `types.ts`
+- Initial commit
+
+## `componentLibrary/fontAwesomev6Icon`
+
+## [0.5.0](https://github.com/code-dot-org/code-dot-org/pull/62803)
+
+- rewritten `FontAwesomeV6IconTest` in typescript
+- refactored/optimized `FontAwesomeV6IconTest.tsx`
+
+## [0.4.0](https://github.com/code-dot-org/code-dot-org/pull/61754)
+
+- updated `FontAwesomeV6Icon` props to support native HTML Element attributes
+
+## [0.3.1](https://github.com/code-dot-org/code-dot-org/pull/61647)
+
+- added new custom icons support
+- updated custom icons story to show new custom icons
+
+## [0.3.0](https://github.com/code-dot-org/code-dot-org/pull/61296)
+
+- added support of `iconFamily` which allows to use different icon families (e.g. brands, duotone, custom icons, etc)
+- updated stories to show how to use Brands, Duotone, and custom icons
+
+## [0.2.2](https://github.com/code-dot-org/code-dot-org/pull/60788)
+
+- update stories to show how to use fa-brands or any other custom families icons
+
+## [0.2.1](https://github.com/code-dot-org/code-dot-org/pull/58469)
+
+- added support of all `aria-` attributes by `FontAwesomeV6Icon` component
+
+## [0.2.0](https://github.com/code-dot-org/code-dot-org/pull/57228)
+
+- added font awesome icon animation support for `FontAwesomeV6Icon` component
+
+## [0.1.1](https://github.com/code-dot-org/code-dot-org/pull/55797)
+
+- minor documentation updates
+
+## [0.1.0](https://github.com/code-dot-org/code-dot-org/pull/55305)
+
+- Initial commit
+- added fontAwesomeV6Icon component
+- added storybook
+- added tests
+
+## `componentLibrary/typography`
+
+## [2.0.2](https://github.com/code-dot-org/code-dot-org/pull/62837)
```

Comment:
Not blocking but this ordering is confusing me. The versions are descending then jump up to 2.0.2 and descend again. Should these be moved to the top of this file so all the versions are strictly descending?

Reply:
Good question!
This Changelog-pre1.md will be a place where all previous Changelogs will be combined into 1, sorted per component. (Because initially we had a separate Changelog for every component).
pre-1 versions of the package might also end up in this file (on the top of the file).
Starting from v1.0.0.alpha0 version all the changelogs will be in a separate, main CHANGELOG.md so that those versions will be sorted in regular pattern everyone uses everywhere.

Also worth mentioning here that Typography is the only component that has v1 and v2 (major versions) since it was the first component we made and since it was used slightly different